File: Leetcode/C++/461. Hamming Distance/solution.cpp
// https://leetcode.com/problems/hamming-distance/submissions/981599269/

class Solution
{
public:
    int hammingDistance(int x, int y)
    {
        int count = 0, result = x ^ y;
        while (result > 0)
        {
            count += result % 2; // When LSB(Least Significant Bit) is 1 add 1 else add 0
            result >>= 1;
        }
        return count;
    }
};

File: Leetcode/Typescript/461. Hamming Distance/solution.ts
// https://leetcode.com/problems/hamming-distance/description/

function hammingDistance(x: number, y: number): number {
  let num = x ^ y;
  let count = 0;

  while (num > 0) {
    count += 1;
    num &= num - 1;
  }
  return count;
}
